### PR TITLE
Update issue templates

### PR DESCRIPTION
**Pull request by dwelch-r7**
Updating the issue templates to use the new format, a lot of inspiration taken from https://github.com/gatsbyjs/gatsby

It would be super nice if we had a few more links/resources we could point at like they do at gatsby but I think this is a decent enough start for now

You can see what the new changes will look like here:
https://github.com/dwelch-r7/msf-issue-templates/issues/new/choose

image

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+name: Feature Request 💡
+about: Suggest a new idea for the project.
+labels: "type: feature or enhancement"
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/feature_request.md:4]: Does this need changed? 👀

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+name: Feature Request 💡
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/feature_request.md:2]: Discussion: Is it worth splitting out generic feature requests vs module requests?

Comment (adfoster-r7) [reply .github/ISSUE_TEMPLATE/feature_request.md:2]: Or maybe we can just modify this to be:

```suggestion
name: Feature / Module Request 💡
about: Suggest a new idea of module for the project.
```

**Comment by dwelch-r7** [comment]
https://github.com/rapid7/metasploit-framework/wiki/Reporting-a-Bug
Dropping this here to remind myself to add the link in later

**Comment by adfoster-r7** [comment]
@dwelch-r7 I was only 20% trolling when I was suggesting to add a high level `termux` issue template 😅

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -1,3 +1,18 @@
...
+<!--
+  Please fill out each section below, otherwise, your issue will be closed. This info allows Metasploit maintainers to diagnose (and fix!) your issue as quickly as possible.
+
+  Useful Links:
+  - Wiki: https://github.com/rapid7/metasploit-framework/wiki
+
+  Before opening a new issue, please search existing issues: https://github.com/rapid7/metasploit-framework/issues
+-->
+
 ## Steps to reproduce
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/bug_report.md:16]: Would it be worthwhile adding a section for what tutorial/documentation are you following?

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (adfoster-r7) [reply .github/ISSUE_TEMPLATE/feature_request.md:2]: On reflection, I think I like the idea of a separate module request section more than combining this into one

File: .github/ISSUE_TEMPLATE/documentation.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+name: Documentation 📝
+about: Suggest better docs coverage for a particular tool or process.
+labels: "type: documentation"
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/documentation.md:4]:
```suggestion
labels: "docs"
```

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 1 above)
Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/feature_request.md:4]: Maybe...I'd like to keep the labels for issues separate from what we label our PRs though, unless you think we should keep them the same?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/bug_report.md:16]: Couldn't hurt!

File: .github/ISSUE_TEMPLATE/documentation.md (hunk 4 above)
Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/documentation.md:4]: yea I didn't want it to be the same as the one we use for "this module has docs" but if you think they should be the same I can do that for sure

File: .github/ISSUE_TEMPLATE/question.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+name: Question 🤔
+about: Usage question or discussion about Metasploit.
+labels: "type: question or discussion"
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/question.md:4]: This label eneds updated I think https://github.com/rapid7/metasploit-framework/labels?q=question

```suggestion
labels: "question"
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 6]
```diff
@@ -0,0 +1 @@
+blank_issues_enabled: false
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Gatsby has nice extra links that appear at the bottom of the issues creation page:

image

It would be cool to update this with the docs site once that comes along. I'm not sure the Wiki is worth linking to directly with the current offerings. However we could potentially have a high level links for Termux? Slack? Common Windows questions?

Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/config.yml:1]: yea we for sure could/should I spent a bit too long searching through looking for things we can link to but there was nothing that really made sense, definitely something for the future

**Comment by gwillcox-r7** [comment]
Looks good to me, only thing I'd say is the question template is a little off imho as you have this line in there:

```
  Before opening a new issue, please search existing issues https://github.com/rapid7/metasploit-framework/issues
```

But I thought that template was for a question not for an issue? I get the intention of not opening questions that have already been asked but perhaps you could reword this to be a bit clearer?

File: .github/ISSUE_TEMPLATE/documentation.md (hunk 4 above)
Comment (adfoster-r7) [reply .github/ISSUE_TEMPLATE/documentation.md:4]: Hm, I guess `docs` as it exists at the minute is actually `has-docs` or something else. I think the existing use of flags is arbitrary, other than the `rn-*` flags which serve a purpose in release notes automation.

I've no strong preference here,  potentially we could go with:
- `docs-request`
- `module-request`
- `feature-request`

Or am I happy to stick with `docs` for issues and pull requests for simplicity/consistency.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/feature_request.md:2]: I've discovered a super major blocker with having two separate templates...what emoji do we use for the module request one?

Comment (adfoster-r7) [reply .github/ISSUE_TEMPLATE/feature_request.md:2]: 📦

File: .github/ISSUE_TEMPLATE/documentation.md (hunk 4 above)
Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/documentation.md:4]: I think the request prefix is a good shout

**Comment by dwelch-r7** [comment]
> Looks good to me, only thing I'd say is the question template is a little off imho as you have this line in there:
> 
> ```
>   Before opening a new issue, please search existing issues https://github.com/rapid7/metasploit-framework/issues
> ```
> 
> But I thought that template was for a question not for an issue? I get the intention of not opening questions that have already been asked but perhaps you could reword this to be a bit clearer?

hmmm not sure about this one, yes it's a question but the question gets dropped into the issue queue the same as everything else, there is no "question" queue the user could search (they could of course filter the issue queue by the question label though)

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (dwelch-r7) [reply .github/ISSUE_TEMPLATE/feature_request.md:2]: I like it

**Comment by gwillcox-r7** [comment]
> > Looks good to me, only thing I'd say is the question template is a little off imho as you have this line in there:
> > ```
> >   Before opening a new issue, please search existing issues https://github.com/rapid7/metasploit-framework/issues
> > ```
> > 
> > 
> > But I thought that template was for a question not for an issue? I get the intention of not opening questions that have already been asked but perhaps you could reword this to be a bit clearer?
> 
> hmmm not sure about this one, yes it's a question but the question gets dropped into the issue queue the same as everything else, there is no "question" queue the user could search (they could of course filter the issue queue by the question label though)

Hmm perhaps it might be better to say this then? ```Before opening a new issue, please search the issue tracker at https://github.com/rapid7/metasploit-framework/issues to ensure there are no existing issues which answer your question.```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+about: Suggest a new idea for the project.
+labels: "request-feature"
```
Comment (adfoster-r7) [inline .github/ISSUE_TEMPLATE/feature_request.md:4]: Are we renaming the existing label, or creating a new one?

https://github.com/rapid7/metasploit-framework/labels?q=feature

**Review by adfoster-r7 — APPROVED** [review]
